packit/utils/obs_helper.py

```diff
@@ -0,0 +1,454 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import logging
+import os
+import re
+import xml.etree.ElementTree as ET
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+from osc import conf, core
+
+from packit.config import PackageConfig
+from packit.config.aliases import DEPRECATED_TARGET_MAP
+
+logger = logging.getLogger(__name__)
+
+_API_URL = "https://api.opensuse.org"
+
+
+@dataclass(frozen=True)
+class XmlPathEntry:
+    """Representation of a path entry in the XML repository configuration of
+    OBS.
+
+
+    """
+
+    project: str
+    repository: str
+
+
+@dataclass(frozen=True)
+class OBSRepository:
+    """Minimal representation of a repository on OBS that is part of the project
+    meta configuration:
+
+    .. code-block:: xml
+
+       <repository name="images">
+         <path project="openSUSE:Factory" repository="containerfile"/>
+         <path project="openSUSE:Factory" repository="standard"/>
+         <arch>x86_64</arch>
+       </repository>
+
+    """
+
+    name: str
+    arch: list[str]
+    path: list[XmlPathEntry]
+
+
+def target_to_path(target: str) -> list[XmlPathEntry]:
+    """Converts a packit target name like ``fedora-rawhide-x86_64`` or
+    ``opensuse-leap-15.5`` to a list of xml path entries of the respective
+    projects on OBS. This function relies on the project setup on
+    build.opensuse.org and is not directly re-usable on other OBS instances.
+
+    """
+
+    target_split = target.split("-")
+    version, arch = target_split[-2:]
+    distro = "-".join(target_split[:-2])
+
+    if distro == "fedora":
+        if version == "rawhide":
+            return [XmlPathEntry(project="Fedora:Rawhide", repository="standard")]
+        return [
+            XmlPathEntry(project=f"Fedora:{version}", repository="standard"),
+            XmlPathEntry(project=f"Fedora:{version}", repository="update"),
+        ]
+
+    if distro == "epel":
+        if version == "9":
+            return [
+                XmlPathEntry(project=f"Fedora:EPEL:{version}", repository="stream"),
+            ]
+        if version in ("8", "7"):
```

Comment:
> Just an FYI, we already have 10

Why this would be a good approach to supporting new versions, OBS doesn't version 10 at this moment...

> Also I'm kind of confused why you're returning `repository=stream` for EPEL (this is not tied _exclusively_ to the CentOS)

This is basically how OBS defines the repository. There's really nothing we can do about it.